html/js/alpine.min.js

```diff
@@ -0,0 +1,5 @@
hasAttribute(a)&&c===null?l():s.hasAttribute(a)?(u(),l()):u()}i.forEach((o,s)=>{Fe(s,o)}),n.forEach((o,s)=>{bt.forEach(a=>a(s,o))});for(let o of t)r.includes(o)||wt.forEach(s=>s(o));for(let o of r)t.includes(o)||vt.forEach(s=>s(o));t=null,r=null,n=null,i=null}function V(e,t,r){return e._x_dataStack=[t,...Z(r||e)],()=>{e._x_dataStack=e._x_dataStack.filter(n=>n!==t)}}function He(e,t){let r=e._x_dataStack[0];Object.entries(t).forEach(([n,i])=>{r[n]=i})}function Z(e){return e._x_dataStack?e._x_dataStack:e instanceof ShadowRoot?Z(e.host):e.parentNode?Z(e.parentNode):[]}function Be(e){return new Proxy({},{ownKeys:()=>Array.from(new Set(e.flatMap(t=>Object.keys(t)))),has:(t,r)=>e.some(n=>n.hasOwnProperty(r)),get:(t,r)=>(e.find(n=>n.hasOwnProperty(r))||{})[r],set:(t,r,n)=>{let i=e.find(o=>o.hasOwnProperty(r));return i?i[r]=n:e[e.length-1][r]=n,!0}})}function Ct(e){let t=n=>typeof n=="object"&&!Array.isArray(n)&&n!==null,r=(n,i="")=>{Object.entries(n).forEach(([o,s])=>{let a=i===""?o:`${i}.${o}`;typeof s=="object"&&s!==null&&s._x_interceptor?n[o]=s.initialize(e,a,o):t(s)&&s!==n&&!(s instanceof Element)&&r(s,a)})};return r(e)}function de(e,t=()=>{}){let r={initialValue:void 0,_x_interceptor:!0,initialize(n,i,o){return e(this.initialValue,()=>Lr(n,i),s=>qe(n,i,s),i,o)}};return t(r),n=>{if(typeof n=="object"&&n!==null&&n._x_interceptor){let i=r.initialize.bind(r);r.initialize=(o,s,a)=>{let c=n.initialize(o,s,a);return r.initialValue=c,i(o,s,a)}}else r.initialValue=n;return r}}function Lr(e,t){return t.split(".").reduce((r,n)=>r[n],e)}function qe(e,t,r){if(typeof t=="string"&&(t=t.split(".")),t.length===1)e[t[0]]=r;else{if(t.length===0)throw error;return e[t[0]]||(e[t[0]]={}),qe(e[t[0]],t.slice(1),r)}}var Rt={};function x(e,t){Rt[e]=t}function Q(e,t){return Object.entries(Rt).forEach(([r,n])=>{Object.defineProperty(e,`$${r}`,{get(){return n(t,{Alpine:S,interceptor:de})},enumerable:!1})}),e}function v(e,t,r={}){let n;return h(e,t)(i=>n=i,r),n}function h(...e){return Mt(...e)}var Mt=Ue;function Nt(e){Mt=e}function Ue(e,t){let r={};Q(r,e);let n=[r,...Z(e)];if(typeof t=="function")return $r(n,t);let i=Fr(n,t);return jr.bind(null,e,t,i)}function $r(e,t){return(r=()=>{},{scope:n={},params:i=[]}={})=>{let o=t.apply(Be([n,...e]),i);pe(r,o)}}var We={};function Kr(e){if(We[e])return We[e];let t=Object.getPrototypeOf(async function(){}).constructor,r=/^[\n\s]*if.*\(.*\)/.test(e)||/^(let|const)/.test(e)?`(() => { ${e} })()`:e,n=new t(["__self","scope"],`with (scope) { __self.result = ${r} }; __self.finished = true; return __self.result;`);return We[e]=n,n}function Fr(e,t){let r=Kr(t);return(n=()=>{},{scope:i={},params:o=[]}={})=>{r.result=void 0,r.finished=!1;let s=Be([i,...e]),a=r(r,s);r.finished?pe(n,r.result,s,o):a.then(c=>{pe(n,c,s,o)})}}function pe(e,t,r,n){if(typeof t=="function"){let i=t.apply(r,n);i instanceof Promise?i.then(o=>pe(e,o,r,n)):e(i)}else e(t)}function jr(e,t,r,...n){try{return r(...n)}catch(i){throw console.warn(`Alpine Expression Error: ${i.message}
```

Comment:
we should require this via npm instead

Reply:
Sure, but this is just dipping our toes in alpine.js